### DISCUSSION TITLE
Problems Installing MonaiLabel with Proxy

### DISCUSSION THREAD

**Discussion by evaherbst** [Project-MONAI/MONAILabel#1671] [General]
Hi all,

I would like to run MonaiLabel locally to use with slicer. I have CUDA enabled (11.8).

I tried to install MonaiLabel with docker. I am using windows.

I can run docker using:
_docker run --gpus all --rm -ti --ipc=host --net=host projectmonai/monailabel:latest bash_

and start the monai server, and I get:
`_Uvicorn running on http://0.0.0.0:8000_`

but when I try to access it with MonaiLabel in 3D slicer I get this issue:
I tried using hosts:
http://0.0.0.0:8000/ or
http://127.0.0.1:8000/ or
http://localhost:8000/ or leaving it blank

details:
ConnectionRefusedError: [WinError 10061] No connection could be made because the target machine actively refused it

I also get an "access denied" error when I enter the http://0.0.0.0:8000/ in a browser.

My organisation uses a proxy server to connect to the internet. Do I somehow need to change the label server address to account for this?

I also saw in docker I can adjust the ports:
![image](https://github.com/Project-MONAI/MONAILabel/assets/31034865/a15da2ca-d224-499b-aa08-0cb4c5e60552)

Or is there maybe an issue with ports because of WSL being used by docker?

thank you so much,
Eva

**Comment by SachidanandAlle** [comment]
You can also use the pip install monailabel option on Windows.. as you are planning to use 3D slicer as the client

**Reply by evaherbst** [reply]
Thanks for the feedback!
I also tried this with an anaconda venv and directly and also get issues.

I checked “nvcc-version” and cuda 11.8 is installed so I used:
`pip install torch torchvision torchaudio --extra-index-url https://download.pytorch.org/whl/cu118
`
however, when I run “python -c “import torch; print(torch.cuda.is_available())”” it just stalls.

If I skip checking this and start downloading the monailabel apps, it also freezes:
image

And if I install directly without venv I get a different error

```
NAILabelReviewer\UnitTests\TestDataSet
error: could not create ‘build\bdist.win-amd64\wheel\monailabel-0.8.2rc2.data\data\monailabel\plugins\slicer\MONAILabelReviewer\UnitTests\TestDataSet\test_datastore_v2_image_without_segmentation.json’: No such file or directory
[end of output]

note: This error originates from a subprocess, and is likely not a problem with pip.
ERROR: Failed building wheel for monailabel
Failed to build monailabel

```

**Reply by SachidanandAlle** [reply]
Try direct python version (3.10) from python.com.. also you need cuda enabled correctly irrespective of python/conda/docker..

**Reply by evaherbst** [reply]
thanks I will try python 3.10. I have cuda 11.8 installed

**Reply by evaherbst** [reply]
Hello. I have tried installing with python 3.10.
I abandoned the docker route for now and trying to install using a conda virtual environment:

I tried:
```
conda create -n monai python=3.10 -y
conda install pytorch torchvision torchaudio pytorch-cuda=11.8 -c pytorch -c nvidia
pip install monailabel
```
those worked, then I ran
monailabel apps

**As before with my conda venv installation, this resulted in: **
`Using PYTHONPATH=C:\Users\Eva\anaconda3\envs;`
and then nothing else happening.
and 
`python -c "import torch; print(torch.cuda.is_available())"`
just stalls

Previous posts ([here](https://github.com/Project-MONAI/MONAILabel/issues/1469) and [here](https://github.com/Project-MONAI/MONAILabel/discussions/1130)) have identified that this could be an issue with using "python3" vs "python" and solved it with:
`python-is-python3`

However, this does not work on Windows. I also tried mlink but mlink is not recognized by my machine.

Does someone have a solution how to do this on windows?

**Reply by diazandr3s** [reply]
Hi @evaherbst,

Thanks for the update. 

> **As before with my conda venv installation, this resulted in: **
> Using PYTHONPATH=C:\Users\Eva\anaconda3\envs;
> and then nothing else happening.

Can you please provide more details of the error you're getting? Maybe the full logs from the terminal?

> 
> Previous posts (here and here) have identified that this could be an issue with using "python3" vs "python" and solved it with:
> python-is-python3

Have you tried the solution proposed here? https://gist.github.com/troyfontaine/e1174ffa2140e7d6f092fe0984225a1d

**Reply by evaherbst** [reply]
Thank you Andres!
I get no error info. The console just returns _PYTHONPATH=C:\Users\Eva\anaconda3\envs;_
When I run
monailabel apps

and it returns nothing and does nothing when I try

_python -c "import torch; print(torch.cuda.is_available())"_

I will also try the solution you proposed

**Reply by evaherbst** [reply]
Thank you @diazandr3s 
I am finally at the clinic again. I tried your solution, setting both alias for python3 to link to python and vice versa. When I check where python in my anaconda venv, it points to python.exe, and I checked this exists (the issues from other usersseemed to only have python3). I have not been able to fix the issue with the alias solution. 

The anaconda prompt  just returns PYTHONPATH=C:\Users\Eva\anaconda3\envs; "" when I try to run "monailabel apps". Somehow the cuda.is_available check works now and returns true.

Could someone confirm that the right python path is being used?

I see the code for printing the pythonpath is in the [monailabel.bat file ](https://github.com/Project-MONAI/MONAILabel/blob/747f1d9802359a1a26b068357fe952371ff63c20/monailabel/scripts/monailabel.bat)

The PythonPath it prints is C:\Users\Eva\anaconda3\envs for me.
The python.exe is at C:\Users\Eva\anaconda3\envs\MonaiLabel\python.exe
So is monailabel pointing to an incorrect PhythonPath? 

I tried hard coding the PythonPath in the monailbael.bat file to 
set PYTHONPATH="C:\Users\Eva\anaconda3\envs\MonaiLabel"
or
set PYTHONPATH=C:\Users\Eva\anaconda3\envs\MonaiLabel\python.exe 
but they both stall as well.

Here is another image of the problem and the relevant code in monailabel.bat:
The console prints PythonPath= and then prints "" and then stalls (maybe it is not breaking, but rather being incredibly slow - nothing happens for over 10 minutes)

![image](https://github.com/Project-MONAI/MONAILabel/assets/31034865/04ec7097-609e-4dfc-bfe8-9499184f60ee)

**### Updates:**
**I also tried with miniconda. Still the same issue (lagging, nothing happens).**

**so to summarize all of the issues:**
- **With install from github I get the error:** " Could not build wheels for monailabel, which is required to install pyproject.toml-based projects"

as well as several warnings like this: "7: _Warning: Package 'monailabel.scripts' is absent from the `packages` configuration.
      !!"

-**With pip install monailabel I get the lagging.**

**-With direct install on computer (no conda venv)**
I the following error (note I printed out my cuda and torch versions, they are compatible)
![image](https://github.com/Project-MONAI/MONAILabel/assets/31034865/4ec00ef6-2b88-41f3-9436-c5953141ebc9)

**Comment by evaherbst** [comment]
The issue is resolved.

I am not sure why (since I was running MonaiLabel locally) but it seems that the cause of my installation issues was due to the university proxy (which I was told to use to access the internet).
I instead purchased a WIFI dongle and used WIFI (to avoid the proxy). I installed monailabel in a miniconda venv on Windows and everything worked well.
